neovim now depends on gperf for building

As of neovim/neovim@7e2348f2b1b487c875bbcf6c6711a276f9063040 neovim requires `gperf` during build.

The following doesn't seem to work because `gperf` is not in homebrew-core.

```ruby
    resource "gperf" do
      url "https://ftpmirror.gnu.org/gperf/gperf-3.0.4.tar.gz"
      sha256 "767112a204407e62dbc3106647cf839ed544f3cf5d0f0523aaa2508623aad63e"
    end
```

Setting `USE_EXISTING_SRC_DIR` to `OFF` fixes the build but I'm not sure if that is the proper approach. What would the proper approach be?

I'm getting the following error:
```console
Last 15 lines from /Users/kborges/Library/Logs/Homebrew/neovim/02.make:
/usr/local/Cellar/cmake/3.6.1/bin/cmake -E make_directory /tmp/neovim-20160902-70131-dpc1hs/deps-build/build/tmp
/usr/local/Cellar/cmake/3.6.1/bin/cmake -E make_directory /tmp/neovim-20160902-70131-dpc1hs/deps-build/build/src/gperf-stamp
/usr/local/Cellar/cmake/3.6.1/bin/cmake -E make_directory /tmp/neovim-20160902-70131-dpc1hs/deps-build/build/downloads/gperf
/usr/local/Cellar/cmake/3.6.1/bin/cmake -E touch /tmp/neovim-20160902-70131-dpc1hs/deps-build/build/src/gperf-stamp/gperf-mkdir
[ 92%] Performing download step for 'gperf'
cd /tmp/neovim-20160902-70131-dpc1hs/deps-build/build/downloads/gperf && /usr/local/Cellar/cmake/3.6.1/bin/cmake -DPREFIX=/tmp/neovim-20160902-70131-dpc1hs/deps-build/build -DDOWNLOAD_DIR=/tmp/neovim-20160902-70131-dpc1hs/deps-build/build/downloads/gperf -DURL=http://ftp.gnu.org/pub/gnu/gperf/gperf-3.0.4.tar.gz -DEXPECTED_SHA256=767112a204407e62dbc3106647cf839ed544f3cf5d0f0523aaa2508623aad63e -DTARGET=gperf -DUSE_EXISTING_SRC_DIR=ON -P /tmp/neovim-20160902-70131-dpc1hs/third-party/cmake/DownloadAndExtractFile.cmake
CMake Error at /tmp/neovim-20160902-70131-dpc1hs/third-party/cmake/DownloadAndExtractFile.cmake:33 (message):
  USE_EXISTING_SRC_DIR set to ON, but
  '/tmp/neovim-20160902-70131-dpc1hs/deps-build/build/src/gperf' does not
  exist or is empty.


make[2]: *** [build/src/gperf-stamp/gperf-download] Error 1
make[1]: *** [CMakeFiles/gperf.dir/all] Error 2
make: *** [all] Error 2
```

--- a/Formula/neovim.rb
+++ b/Formula/neovim.rb
@@ -119,7 +119,8 @@ class Neovim < Formula
     cd "deps-build" do
       ohai "Building third-party dependencies."
       system "cmake", "../third-party", "-DUSE_BUNDLED_BUSTED=OFF",
-             "-DUSE_BUNDLED_LUV=OFF", "-DUSE_EXISTING_SRC_DIR=ON",
+             "-DUSE_BUNDLED_GPERF=OFF", "-DUSE_BUNDLED_LUV=OFF",
+             "-DUSE_EXISTING_SRC_DIR=ON",
              *std_cmake_args
       system "make", "VERBOSE=1"
     end

